contracts/FeeCustody.sol

```diff
@@ -0,0 +1,405 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "./interfaces/IFeeDistributor.sol";
+import "./interfaces/IChainlink.sol";
+
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol";
+
+/** @title FeeCustody
+    @notice Custody Contract for Ribbon Vault Management / Performance Fees
+ */
+
+contract FeeCustody is Ownable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    // Distribution token for fee distributor: like RBN, USDC, ETH, etc
+    IERC20 public distributionToken;
+    // Protocol revenue recipient
+    address public protocolRevenueRecipient;
+    // Address of fee distributor contract for RBN lockers to claim
+    IFeeDistributor public feeDistributor;
+
+    // % allocation (0 - 100%) from protocol revenue to allocate to RBN lockers.
+    // 2 decimals. ex: 10% = 1000
+    uint256 public pctAllocationForRBNLockers;
+
+    uint256 public constant TOTAL_PCT = 10000; // Equals 100%
+    ISwapRouter public constant UNIV3_SWAP_ROUTER =
+        ISwapRouter(0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45);
+
+    mapping(address => bool) public assetExists;
```

Comment:
`assetExists` feels redundant because you can check if the asset exists using `intermediaryPath`. Since `intermediaryPath` for an asset needs to exist even if it's a single hop